jwst/outlier_detection/outlier_detection.py

```diff
@@ -291,8 +291,16 @@ def create_median(self, resampled_models):
         # Compute median of stack of images using `badmasks` to remove
         # low-weight values.  In the future we should use a masked array
         # and np.median
-        median_image = median(resampled_sci, nlow=nlow, nhigh=nhigh,
-                              badmasks=badmasks)
+        max_images_for_median = 1024
```

Comment:
Looking at the comment above, we could avoid using `stsci.image.median` completely by just using `np.median` and using a masked array.  Or by turning the masked values into nans and using `np.nanmedian`.  Either would work, and would use all the data to create the median.  Using `np.nanmedian` might take less memory and be faster.

The only reason we are using `stsci.image.median` is for the case when there are several images in the stack, and one _might_ want to toss out the high and low values before doing the median, though we don't use this mode at all in pipeline processing.

Or one could take a hybrid approach and use `stsci.image.median` for cases were n <= 1024, and just do a straight median using `np.nanmedian` for the >1024  case, as there's no chance throwing out a few high or low cases will matter at all if there are >1024 images in the stack.

That said, even if we are using plain `np.nanmedian` here, on a stack of 3000 integrations, the memory footprint would be very high for a 2048x2048 image. I assume we are not using full arrays here but subarrays?

Reply:
From my point of view I'd ask what are we trying to accomplish? So to determine a robust median value as a baseline  to look for anomalous values then 1K inputs should be fine. I'm not sure what adding another 50, 100, or 2000 images will accomplish. If the additional images change the results significantly then we have a different problem and that should be addressed separately. So for outlier detection I think just picking the first N images should work well. 

If we need a median image for a large number of images then moving away from stimage might be a good idea. As you say the memory footprint for thousands of 1024^2 images would be an issue.

Reply:
We should never have a situation of a very large number of large images. The current problem case is for a very large stack of 2D spectra, but the 2D cutouts are not terribly big. Still, if we ever get a situation where there are 30-60k of these (which is allowed in the APT), we could be in trouble in terms of memory footprint.

I agree that we should just take this opportunity to switch all cases to using `np.nanmedian` and be done with the limitation imposed in `stsci.image.median`. I also think it's appropriate to limit the number of inputs we use in `np.nanmedian`, because we could get into memory footprint issues in the future if we just blindly try to use all inputs. Setting a limit of 1024 is a reasonable baseline. Can always come back and tweak that if testing shows a different value works better.

Reply:
Can we somehow use the existing `badmasks` to, for example, intentionally set those pixels to NaN so that they won't be used in the `np.nanmedian` calculation? Or use them to create a masked array to send to `np.nanmedian`.